src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1076,6 +1076,15 @@ static inline bool notification_match(reservation_t& res,
   return true;
 }
 
+static inline uint64_t compute_target_shard(const DoutPrefixProvider* dpp, const std::string& bucket_name, const std::string& object_key, const uint64_t& num_shards){
```

Comment:
no need to pass `unit64_t` as const ref.
also, assuming the number of shards is not too big, you can probably use a uint32 or even uint16